tests/protocol_test.py

```diff
@@ -0,0 +1,39 @@
+"""Tests for the protocol client."""
+
+import unittest
+import uuid
+
+from api.xline.xline_command_pb2 import Command, RequestWithToken
+from api.xline.rpc_pb2 import PutRequest
+from client.protocol import ProtocolClient
+
+
+class TestProtocolClient(unittest.TestCase):
+    """test protocol client"""
+
+    def setUp(self) -> None:
+        curp_members = ["172.20.0.3:2379", "172.20.0.4:2379", "172.20.0.5:2379"]
+
+        cmd = Command(
+            request=RequestWithToken(
+                put_request=PutRequest(
+                    key=b"hello",
+                    value=b"py-xline",
+                )
+            ),
+            propose_id=f"client-{uuid.uuid4()}",
+        )
+
+        self.cmd = cmd
+        self.client = ProtocolClient.build_from_addrs(curp_members)
```

Comment:
IMO, we'd better mock a protocol client for testing. Depending on a real protocol client in unit tests is not a good practice.

Reply:
We'll implement it later to provide mock functionality for all language clients.